src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncSpillTests.cs

```diff
@@ -8185,210 +8274,1361 @@ static async Task Main()
+                                int V_2,
+                                int V_3)
+                  IL_0000:  ldarg.0
+                  IL_0001:  ldobj      "S"
+                  IL_0006:  stloc.0
+                  IL_0007:  ldloc.0
+                  IL_0008:  ldfld      "int S.i"
+                  IL_000d:  stloc.1
+                  IL_000e:  call       "System.Threading.Tasks.Task<int> S.GetInt()"
+                  IL_0013:  call       "int System.Runtime.CompilerServices.AsyncHelpers.Await<int>(System.Threading.Tasks.Task<int>)"
+                  IL_0018:  stloc.2
+                  IL_0019:  ldloca.s   V_0
+                  IL_001b:  ldloc.1
+                  IL_001c:  ldloc.2
+                  IL_001d:  add
+                  IL_001e:  dup
+                  IL_001f:  stloc.3
+                  IL_0020:  stfld      "int S.i"
+                  IL_0025:  ldloc.3
+                  IL_0026:  call       "void System.Console.Write(int)"
+                  IL_002b:  ret
+                }
+                """);
+        }
+
+        [Fact]
+        public void SpillCompoundAssignmentToThisStruct_02()
+        {
+            var source = """
+                using System;
+                using System.Threading.Tasks;
+                struct S
+                {
+                    int i;
+
+                    public static S operator +(S s, int value)
+                    {
+                        s.i += value;
+                        return s;
+                    }
+
+                    async Task M()
+                    {
+                        Console.Write((this += await GetInt()).i);
+                    }
+
+                    static async Task Main()
+                    {
+                        S s = default;
+                        await s.M();
+                        Console.Write(s.i);
+                    }
+
+                    static Task<int> GetInt() => Task.FromResult(1);
+                }
+                """;
+
+            var expectedOutput = "10";
+            CompileAndVerify(source, expectedOutput: expectedOutput, options: TestOptions.ReleaseExe);
+            CompileAndVerify(source, expectedOutput: expectedOutput, options: TestOptions.DebugExe);
+
+            var comp = CreateRuntimeAsyncCompilation(source);
+            var verifier = CompileAndVerify(comp, expectedOutput: CodeGenAsyncTests.ExpectedOutput(expectedOutput, isRuntimeAsync: true), verify: Verification.Fails with
+            {
+                ILVerifyMessage = """
+                    [M]: Return value missing on the stack. { Offset = 0x21 }
+                    [Main]: Return value missing on the stack. { Offset = 0x1f }
+                    """
+            });
+
+            verifier.VerifyDiagnostics();
+            verifier.VerifyIL("S.M()", """
+                {
+                  // Code size       34 (0x22)
+                  .maxstack  2
+                  .locals init (int V_0)
```

Comment:
I found it a bit surprising that 'this' was not hoisted to a temp here. It looks like instead we just use the result of the `S operator +(S, int)` and access a field on it? Maybe the stack optimizer pass eliminated the temp?